pkg/policy/api/v3/v2tov3_test.go

```diff
@@ -0,0 +1,322 @@
+		name string
+		args args
+		want *Rule
+	}{
+	// TODO: Add test cases.
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := V2RuleTov3Rule(tt.args.v2Rule); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("V2RuleTov3Rule() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_v2ESTov3ES(t *testing.T) {
+	type args struct {
+		v2ES *v2.EndpointSelector
+	}
+	tests := []struct {
+		name string
+		args args
+		want *IdentitySelector
+	}{
+	// TODO: Add test cases.
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := v2ESTov3ES(tt.args.v2ES); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("v2ESTov3ES() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_v2IRTov3IR(t *testing.T) {
+	type args struct {
+		v2IR *v2.IngressRule
+	}
+	tests := []struct {
+		name string
+		args args
+		want []IngressRule
+	}{
+	// TODO: Add test cases.
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := v2IRTov3IR(tt.args.v2IR); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("v2IRTov3IR() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_v2ERTov3ER(t *testing.T) {
+	type args struct {
+		v2ER *v2.EgressRule
+	}
+	tests := []struct {
+		name string
+		args args
+		want []EgressRule
+	}{
+	// TODO: Add test cases.
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := v2ERTov3ER(tt.args.v2ER); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("v2ERTov3ER() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_v2CIDRRuleTov3CIDRRule(t *testing.T) {
+	type args struct {
+		v2CR *v2.CIDRRule
+	}
+	tests := []struct {
+		name string
+		args args
+		want *CIDRRule
+	}{
+	// TODO: Add test cases.
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := v2CIDRRuleTov3CIDRRule(tt.args.v2CR); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("v2CIDRRuleTov3CIDRRule() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_v2CIDRTov3CIDR(t *testing.T) {
+	type args struct {
+		v2C v2.CIDR
+	}
+	tests := []struct {
+		name string
+		args args
+		want CIDR
+	}{
+	// TODO: Add test cases.
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := v2CIDRTov3CIDR(tt.args.v2C); got != tt.want {
+				t.Errorf("v2CIDRTov3CIDR() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_v2EntityTov3Entity(t *testing.T) {
+	type args struct {
+		v2E *v2.Entity
+	}
+	tests := []struct {
+		name string
+		args args
+		want *Entity
+	}{
+	// TODO: Add test cases.
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := v2EntityTov3Entity(tt.args.v2E); got != tt.want {
+				t.Errorf("v2EntityTov3Entity() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_v2PRTov3PR(t *testing.T) {
+	type args struct {
+		v2PR *v2.PortRule
+	}
+	tests := []struct {
+		name string
+		args args
+		want *PortRule
+	}{
+	// TODO: Add test cases.
```

Comment:
I'm not for merging this before we add extensive testing to be sure we translate rules effectively.